dart/lib/vm_service_lib.dart

```diff
@@ -1295,6 +1306,23 @@ class BoundVariable {
     scopeEndTokenPos = json['scopeEndTokenPos'];
   }
 
+  Map<String, dynamic> toJson() {
+    var json = <String, dynamic>{};
```

Comment:
You might instead gen the `toJson()` methods like this:

```
var result = <String, dynamic>{
  'foo': foo,
  'bar': bar.toJson(),
};

if (baz != null) {
  result['baz] = baz.toJson();
}

return result;
```

Which would make the generated method more compact, still handle the optional fields, and possibly make the methods more readable. It is generated code, but we've made an effort here to try and emit something close to what people would write by hand.

Reply:
Ya the code this generates right now is pretty ugly. The only reason I didn't do it that way is it complicates the generating code some, and I just was checking to see if I hit any unexpected edge cases first :).

Happy to do that for the final version though (maybe ill just create an extra string buffer for the optional fields and append at the end so it only takes one pass through the fields).

Reply:
Oh t his is also somewhat complicated because of the super calls, if there is a super class we need to call toJson on it first and then append to the result of that. The existing code path works pretty seamlessly with that, but I could just  use `addAll` and still generate a map literal.

Reply:
Collection spreads and conditional elements will be nice here :)

Reply:
The new collections stuff will definitely be nice, but for now I included a helper method which cleans things up a bunch I think for the optional fields.